packages/ruleset/src/utils/check-subschemas-for-prop.js

```diff
@@ -1,36 +1,15 @@
-const checkSubschemasForProperty = (schema, name) => {
-  if (!schema) {
-    return false;
-  }
-
-  let propertyIsDefined = false;
+const checkCompositeSchemaForConstraint = require('./check-composite-schema-for-constraint');
+const isObject = require('./is-object');
 
-  // first check the properties
-  if (schema.properties) {
-    propertyIsDefined = name in schema.properties;
-  } else if (schema.oneOf || schema.anyOf) {
-    // every schema in a oneOf or anyOf must contain the property
-    const subschemas = schema.oneOf || schema.anyOf;
-    if (Array.isArray(subschemas)) {
-      propertyIsDefined = true;
-      for (const s of subschemas) {
-        if (!checkSubschemasForProperty(s, name)) {
-          propertyIsDefined = false;
-          break;
-        }
-      }
-    }
-  } else if (Array.isArray(schema.allOf)) {
-    // at least one schema in an allOf must contain the property
-    for (const s of schema.allOf) {
-      if (checkSubschemasForProperty(s, name)) {
-        propertyIsDefined = true;
-        break;
-      }
-    }
-  }
-
-  return propertyIsDefined;
+/**
+ * This function will return `true` if all possible variations of a (possibly) composite schema
```

Comment:
Something about this wording seems not quite right to me, since the schema may just have properties and not be a "composite" schema. Maybe something like
```suggestion
 * This function will return `true` if a schema, or all possible variations of a (possibly) composite schema,
```

Do you know what I mean? Feel free to disagree. I don't know if my wording really accomplishes much, it just made me scratch my head for a second that this function would be specifically focused first and foremost on composite schemas (even if the composite-ness was only a "possibility" 🙂 ).

Or maybe even just
```suggestion
 * This function will return `true` if all possible variations of a (possibly composite) schema
```